WordPress/UITestsFoundation/Screens/Login/WelcomeScreen.swift

```diff
@@ -31,6 +33,6 @@ public class WelcomeScreen: BaseScreen {
     }
 
     static func isLoaded() -> Bool {
-        return XCUIApplication().buttons[ElementStringIDs.loginButton].exists
+        (try? WelcomeScreen().isLoaded) ?? false
     }
```

Comment:
I know this is the same in previous PRs, but for some reason it caught my eye only now. 

To be honest, I spent some time not understanding the reason for having an `isLoaded()` method for some of the screens, when there's an `isLoaded` variable in the `ScreenObject`. After a bit of processing, I realized that this is a shorter way to calculate that property in the runtime, without the need to handle potential error every time.

Reply:
Correct.

To be fair, the `static` approach made sense only in the `BaseScreen` context, where a calling `init` on a screen not displayed would fail the test. With `ScreenObject`, it's less useful. I'd like to remove as many of these as possible.

For example, the following doesn't make much sense anymore:

https://github.com/wordpress-mobile/WordPress-iOS/blob/b6d9650a894838f5879ee3d852fda32310171ae7/WordPress/UITestsFoundation/Screens/Editor/BlockEditorScreen.swift#L141-L147

We could replace it with something like:

```swift
guard let alert = FancyAlertComponent() else {
    publishNowButton.tap() 
    return
}

alert.acceptAlert()
```

On the other hand, this does. Since we don't access any of those screen directly, it's handy to call a method on the type:

https://github.com/wordpress-mobile/WordPress-iOS/blob/3675ad4ed9bc44dd374a2b2b10c1fdf6efb587d1/WordPress/WordPressUITests/Flows/EditorFlow.swift#L6-L8

Reply:
Thanks for explaining it, Gio!